python/ml4ir/base/data/ranklib_reader.py

```diff
@@ -0,0 +1,95 @@
+import os
+import pandas as pd
+import tensorflow as tf
+
+from ml4ir.base.io.file_io import FileIO
+from ml4ir.base.features.feature_config import FeatureConfig
+from ml4ir.base.data import tfrecord_reader, tfrecord_writer, ranklib_to_ml4ir
+
+from typing import List
+
+TFRECORD_FILE = "file_0.tfrecord"
+
+def read(
+    data_dir: str,
+    feature_config: FeatureConfig,
+    tfrecord_type: str,
+    tfrecord_dir: str,
+    file_io: FileIO,
+    batch_size: int = 128,
+    preprocessing_keys_to_fns: dict = {},
+    use_part_files: bool = False,
+    max_sequence_size: int = 25,
+    parse_tfrecord: bool = True,
+    logger=None,
+    keep_additional_info = 0,
+    non_zero_features_only = 1,
+    **kwargs
+) -> tf.data.TFRecordDataset:
+    """
+    - reads ranklib-formatted data from an input directory
+    - selects relevant features
+    - creates Dataset X and y
+
+    Current execution plan:
+        1. Conver ranklib to a dataframe
+        2. Convert each query into tf.train.SequenceExample protobufs
+        3. Write the protobufs into a .tfrecord file
+        4. Load .tfrecord file into a TFRecordDataset and parse the protobufs
+
+    Args:
```

Comment:
@mohazahran Can you use the numpy doc format?
https://numpydoc.readthedocs.io/en/latest/format.html